src/forestTypes/binnedTree/binStruct.h

```diff
@@ -70,7 +73,7 @@ namespace fp{
 					int tempMoveObs;
 
 					for(int n = 0; n < fpSingleton::getSingleton().returnNumObservations(); n++){
-						randomObsID = fpSingleton::getSingleton().genRandom(fpSingleton::getSingleton().returnNumObservations());
+						randomObsID = randNum.gen(fpSingleton::getSingleton().returnNumObservations());
```

Comment:
should this be `randNum->gen`?

Reply:
The origional way I envisioned this would be for each thread to pull a random number from the singleton.  This makes results not reproducible.  The way to fix this is to give every tree its own random number generator and seed.  So in the future trees should use randNum.gen not pull from the Singleton.